sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/SchemaPruningSuite.scala

```diff
@@ -621,6 +627,21 @@ abstract class SchemaPruningSuite
     }
   }
 
+  testSchemaPruning("SPARK-38918: nested schema pruning with correlated subqueries") {
```

Comment:
Without this PR, this test failed with `java.lang.RuntimeException: Once strategy's idempotence is broken for batch RewriteSubquery`.

Reply:
Yes it looks like a separate issue with column pruning and subquery rewrite (data source v1 only). I will investigate more.